src/yen/__init__.py

```diff
@@ -106,18 +106,20 @@ def _python_bin_path(python_directory: str) -> str:
         return os.path.join(python_directory, "python", "bin", "python3")
 
 
-def ensure_python(python_version: str) -> tuple[str, str]:
+def ensure_python(python_version: str, force_32bit: bool) -> tuple[str, str]:
     """Checks if given Python version exists locally. If not, downloads it."""
     os.makedirs(PYTHON_INSTALLS_PATH, exist_ok=True)
 
     for python_folder_name in os.listdir(PYTHON_INSTALLS_PATH):
         python_folder = os.path.join(PYTHON_INSTALLS_PATH, python_folder_name)
         if python_folder_name.startswith(python_version):
-            # already installed
-            python_bin_path = _python_bin_path(python_folder)
-            return python_folder_name, python_bin_path
+            if force_32bit and python_folder_name.endswith(SUFFIX_32BIT):
```

Comment:
When using force_32bit the `python_folder_name` does not have a 32bit suffix.
 Steps to reproduce:
1. ensure_python(python_version = 3.10.14, force_32bit=false'`
2. ensure_python(python_version = 3.10.14, force_32bit=true'`
3. The `32bit` is overwrites `64bit` python environment in '.yen_pythons'

Reply:
Well looks like I missed your todo `# TODO: use suffix_32bit here` :)